galsim/chromatic.py

```diff
@@ -1586,41 +1584,61 @@ def __init__(self, *args, **kwargs):
-            # sum is separable if all summands have the same SED.
-            if all([obj.SED == SED1 for obj in args[1:]]):
-                self.separable = True
-                self.SED = SED1
-                self.objlist = [o.copy() for o in args]
-        # if not all the same SED, try to identify groups of summands with the same SED.
-        if not self.separable:
-            # Dictionary of: SED -> List of objs with that SED.
-            SED_dict = {}
-            # Fill in objlist as we go.
-            self.objlist = []
-            for obj in args:
-                # if separable, then add to one of the dictionary lists
-                if obj.separable:
-                    if obj.SED not in SED_dict:
-                        SED_dict[obj.SED] = []
-                    SED_dict[obj.SED].append(obj)
-                # otherwise, just add to self.objlist
-                else:
-                    self.objlist.append(obj.copy())
-            # go back and populate self.objlist with separable items, grouping objs with the
-            # same SED.
+        self.interpolated = any(arg.interpolated for arg in args)
+
+        # We can only add ChromaticObjects together if they're either all SED'd or all non-SED'd
+        dimensionless = all(a.SED.dimensionless for a in args)
+        spectral = all(a.SED.spectral for a in args)
+        if not (dimensionless or spectral):
+            raise ValueError("Cannot add dimensionless and spectral ChromaticObjects.")
+
+        # Sort arguments into inseparable objects and groups of separable objects.  Note that
+        # separable groups are only identified if the constituent objects have the *same* SED even
+        # though a proportional SED is mathematically sufficient for separability.  It's basically
+        # impossible to identify if two SEDs are proportional (or even equal) unless they point to
+        # the same memory, so we just accept this limitation.
+
+        # Each input summand will either end up in norm_dict if it's separable, or in self.objlist
+        # if it's inseparable.
+        SED_dict = {}
+        self.objlist = []
+        for obj in args:
+            if obj.separable:
+                if obj.SED not in SED_dict:
+                    SED_dict[obj.SED] = []
+                SED_dict[obj.SED].append(obj)
+            else:
+                self.objlist.append(obj)
+
+        # If everything ended up in a single SED_dict entry (and self.objlist is empty) then this
+        # ChromaticSum is separable.
+        self.separable = (len(self.objlist) == 0 and len(SED_dict) == 1)
+        if self.separable:
+            the_one_SED = list(SED_dict)[0]
+            self.objlist = SED_dict[the_one_SED]
+            # Since we know that the chromatic objects' SEDs already include all relevant
+            # normalizations, we can just multiply the_one_SED by the number of objects.
+            self.SED = the_one_SED * len(SED_dict[the_one_SED])
```

Comment:
>I suppose they wouldn't get found as separable if they were made as `ChromaticObject(mono_disk) * sed + ChromaticObject(mono_bulge) * sed` 

After the fix to GSObject.SED, this case comes out as separable now too.  Added a test to confirm.